src/CommunityToolkit.Maui/PopupService.cs

```diff
@@ -12,6 +12,8 @@ public class PopupService : IPopupService
 
 	static readonly Dictionary<Type, Type> viewModelToViewMappings = new();
 
+	readonly Stack<WeakReference<Popup>> popupDisplayStack = new();
```

Comment:
Good point! I can fix that

Reply:
Coming back to this, what element of thread-safety were you thinking of? A Popup can only be shown on the UI thread as it needs to create UI. We are also only updating this stack when we receive the Closed event which would come from the Popup, I think this would also need to be on the UI thread?